clustered/common/src/main/java/org/ehcache/clustered/common/messages/EhcacheCodec.java

```diff
@@ -37,46 +34,30 @@
 
   @Override
   public byte[] encodeMessage(EhcacheEntityMessage message) {
-    return marshall(message);
+    if (message.getType() == EhcacheEntityMessage.Type.LIFECYCLE_OP) {
+      return LifeCycleOpCodec.encode((LifecycleMessage) message);
+    } else {
+      return ServerStoreOpCodec.encode((ServerStoreOpMessage) message);
+    }
   }
 
   @Override
   public EhcacheEntityMessage decodeMessage(byte[] payload) throws MessageCodecException {
-    return (EhcacheEntityMessage) unmarshall(payload);
+    ByteBuffer payloadBuf = ByteBuffer.wrap(payload);
+    if (payloadBuf.get() == 1) {
+      return LifeCycleOpCodec.decode(payloadBuf);
```

Comment:
this seems to lack symmetry w.r.t how the first byte signifying the type is coded. Also, payloadBuf.get() == 1 seems to be hardcoded and affecting readability and increasing brittleness,.can we have a const or enum here that specifies the type?.